### PR TITLE
feat: Name for mode

### PR DESCRIPTION
**Pull request by tuanbass**
- Each mode should have a name
- Name will be displayed in the help window

#1

### CODE REVIEW

File: lua/layers/mode.lua [hunk 1]
```diff
@@ -55,9 +56,11 @@ end
 --- thus not recommended.
 ---
 ---@toc_entry   Layers.mode.new()
+---@param name string mode name, will be displayed in the help window
```
Comment (debugloop) [inline lua/layers/mode.lua]:
```suggestion
---@param name string? mode name, will override the configured help window title
```

As the param is optional, we should mark it as such. I'd also prefer a slightly reworded doc string, as this kinda contrasts the configured window title.

File: lua/layers/mode.lua [hunk 2]
```diff
@@ -24,6 +24,7 @@
 ---@class layers.mode
 ---@field _win integer
 ---@field _maps table<layers.mapmode, layers.keymaps>
+---@field _mode_name string
```
Comment (debugloop) [inline lua/layers/mode.lua]:
```suggestion
---@field _mode_name string?
```

Needed so that below usage does not cause a lua_ls diagnostic.

**Review by debugloop — COMMENTED** [review]
Thanks for your contribution! Nice idea and functionally flawless already, but I have a minor thing w.r.t. the doc strings 👍🏻

**Comment by debugloop** [comment]
Ah, and in addition, would you please run `lua require("mini.doc").generate()` and commit the resulting change to also update the help pages inside `doc/` 👍🏻 I really need to create a CI pipeline here...

**Comment by tuanbass** [comment]
I run the mini.doc and committed the results as your suggestion.

**Comment by debugloop** [comment]
Thanks again 🚀

**Pull request merged**